### PR TITLE
Fix deprecated virtiofsd args (go shim only)

### LINKED ISSUE #7111: Go shim passes deprecated options to virtiofsd
Since we switched 1 year ago to the rust virtiofsd, it prints deprecation warnings in the journal when started by the go shim :

```
virtiofsd[3394208]: Use of deprecated flag '-f': This flag has no effect, please remove it
virtiofsd[3394208]: Use of deprecated option format '-o': Please specify options without it (e.g., '--cache auto' instead of '-o cache=auto')
```

Unless we are still supporting the C virtiofsd, which I don't think we do, it is about time to do as suggested.

### PR DESCRIPTION
**Pull request by gkurz**
This series fixes some deprecated arguments being passed to `virtiofsd`,
either by dropping them when they aren't needed anymore or by
converting them to the replacement arguments expected by the rust
virtiofsd.

Fixes #7111

### CODE REVIEW

**Comment by gkurz** [comment]
@pmores

FYI, this is supposed to fix the deprecation warning you're currently seeing with your QEMU driver for the rust shim WIP.
Please give it a try (you just need the configuration TOML file).

**Review by c3d — APPROVED** [review]
With minor changes suggested in the cover letter of second commit…
LGTM, thanks @gkurz

**Comment by gkurz** [comment]
> @pmores
> 
> FYI, this is supposed to fix the deprecation warning you're currently seeing with your QEMU driver for the rust shim WIP. Please give it a try (you just need the configuration TOML file).

I'm dropping this change for the configuration TOML of the rust shim  because it is breaking dragonball CI. This is work for another PR.

File: src/runtime/virtcontainers/virtiofsd.go [hunk 1]
```diff
@@ -187,9 +187,6 @@ func (v *virtiofsd) args(FdSocketNumber uint) ([]string, error) {
 		"--syslog",
 		// cache mode for virtiofsd
 		"-o", "cache=" + v.cache,
-		// disable posix locking in daemon: bunch of basic posix locks properties are broken
-		// apt-get update is broken if enabled
-		"-o", "no_posix_lock",
```
[the remark this answers is not in the document]
Comment (gkurz) [reply src/runtime/virtcontainers/virtiofsd.go:192]: Fixed. Thanks !

**Review by cmaf — APPROVED** [review]
thanks @gkurz

**Pull request merged**